### PR TITLE
Replaces The SM Crush Bonus With a New Mechanic: Gas Coefficient

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
Replaces the SM crush bonus with Gas Coefficient(GC). which is dependent on the moles of each gas in the chamber. Gas Coefficient starts at 1, and alters the engine's radiation output by multiplying it with EER, meaning the effective EER for rad production is EER * GC.

The GC is a weighted sum of the portions of all gasses in the chamber, multiplied by a scaling factor which is (moles per tile / 11333)^2 when below 11333 moles per tile and the square root of (moles per tile / 11333) when above 11333 moles per tile.
The weights of each gas are:
* O2 - 1.5
* CO2 - 1
* Plasma - 4
* N2/N2O - 0.55

The resulting expression is:
GC = (o2_portion * 1.5 + co2_portion * 1 + plasma_portion * 4 + n2_portion * 0.55 + n2o_portion * 0.55) * (moles per tile / 11333)^2 When below 11333 moles per tile and
GC = (o2_portion * 1.5 + co2_portion * 1 + plasma_portion * 4 + n2_portion * 0.55 + n2o_portion * 0.55) * (moles per tile / 11333)^0.5 When above 11333 moles per tile 

So if we had 11333 moles of CO2 in the chamber, and the EER was 7000 our total radiation output would be equivalent to:
GC * EER = 2 * 7000 = 14000 EER

The Gas Coefficient is increased by any amount of any of those gasses, in proportion to their amount. This means that generally more gas leads to a higher GC, but you will still get some boost even when the overall mole count is below the original crush threshold.

Adds per tile total and per gas mole count as well as the current mix's gas coefficient to the supermatter monitoring console.

Anomalies now happen when EER * Gas Coefficient is above 5000.

[This wiki page has all the expected changes to the supermatter wiki page](https://paradisestation.org/wiki/index.php?title=User:MigratingCoconut/gas_coefficient)

## Why It's Good For The Game
Crush is an uninteresting mechanic, just a flat boost to power when a certain amount of gas is in the chamber. It does carry some risk in cases where the engine delams, but usually crush setups mitigate that to begin with just due to how difficult it is for them to be set on fire. In addition, the N2 crush setup offered too much reward for too little risk, being the equivalent of a 7000EER engine without the arcs and gas generation. 
Changing to bonus that is dependent on EER and gives greater bonuses for more dangerous gasses should reward risk better as well as give more room to experiment with interesting gas mixes to try and optimize the different parts of the equation.

## Images of changes
image

## Testing
- Messed around with different gas mixes and observed the expected power boost in the debugger.
- Observed anomaly generation occur according to EER and the crush bonus.
- Made sure all values in the console are correct and are displayed correctly

<hr>

### Declaration
image
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
tweak: Changed the SM crush bonus to depend on EER, the amount of each gas in the chamber.
tweak: Adds the mole count and gas coefficient of the gas mix in the chamber as well as the mole count of each gas to the supermatter monitoring console
/:cl:

### CODE REVIEW

File: code/modules/power/engines/supermatter/supermatter.dm
Comment (Contrabang) [inline code/modules/power/engines/supermatter/supermatter.dm:560]: Why remove this?

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/power/engines/supermatter/supermatter.dm
Comment (Migratingcocofruit) [reply code/modules/power/engines/supermatter/supermatter.dm:560]: This was added to prevent N2 crush from generating power without activation the engine. 
This PR changes the crush bonus to be proportional to EER, so it becomes redundant.

Comment (Contrabang) [reply code/modules/power/engines/supermatter/supermatter.dm:560]: I'm still hesitant on it, if its producing radiation it should be enabled so that admins can properly track logs.

**Comment by Migratingcocofruit** [comment]
Drafting this until anomalies are figured out

File: code/modules/power/engines/supermatter/supermatter.dm
Comment (Migratingcocofruit) [reply code/modules/power/engines/supermatter/supermatter.dm:560]: If EER is above 0 other checks are triggered. If EER is 0 no radiation is produced. The bonus is a multiple of EER, so with 0 EER the bonus is 0. Also, this checked for the gas amount being over the crunch threshold, but with the new behaviour the bonus grows proportionally with the amount of gas rather than cutting out completely once below the threshold.

Comment (Contrabang) [reply code/modules/power/engines/supermatter/supermatter.dm:560]: i still dont like it but i dont give enough of a shit

**Comment by Migratingcocofruit** [comment]
The issues here have been resolved for a while, but due to the PTL revert it puts me over the limit so I'm closing it for now.

**Comment by Migratingcocofruit** [comment]
Singulo Buffs got merged so I'm back under the limit.

**Comment by Migratingcocofruit** [comment]
PTL is merged for real now, so I am one again reopening this PR

**Review by Burzah — APPROVED** [review]
Approved for TGUI.
This needs to be examined by another maintainer or reviewer.

**Comment by Migratingcocofruit** [comment]
Can this be test merged once it's ready rather than merged right away? The changes aren't big, but I would rather give people a chance to break this before it's fully in.

**Comment by Burzah** [comment]
> Can this be test merged once it's ready rather than merged right away? The changes aren't big, but I would rather give people a chance to break this before it's fully in.

Sure I can take care of that.

File: code/modules/power/engines/supermatter/supermatter.dm
Comment (silverplatedelta) [inline code/modules/power/engines/supermatter/supermatter.dm:557]: As all power generation is a function of EER, and 0*anything remains zero, we are good to remove this. If someone finds a way to break it later, it should be easy enough to implement a check.

**Review by silverplatedelta — APPROVED** [review]
Cant comment on anything TGUI, but the DM code looks good to me.

**Review by silverplatedelta — APPROVED** [review]
Looks good to me! Code and gameplay is sound.

**Pull request merged**
